src/scripts/main.js

```diff
@@ -1,3 +1,20 @@
 'use strict';
 
-// write code here
+const allLi = document.querySelector('.tree').querySelectorAll('li');
+
+for (const li of allLi) {
+  if (li.children.length !== 0) {
+    // <li> those haven't elements inside must be ignored
+    const span = document.createElement('span'); // we need just li those have.
+
+    // Create element span and insert it in every <li> we get in this condition
+    li.prepend(span); // span tag has a text like a 'neighbor' sibling
+    span.append(span.nextSibling); // move text inside in span
+
+    span.addEventListener('click', function (e) {
+      e.target.nextSibling.hidden = !e.target.nextSibling.hidden;
```

Comment:
The 'nextSibling' property might not always return the expected element, especially if there are text nodes or comments between elements. It's safer to use the 'nextElementSibling' property.